### PR TITLE
DOC: Make section title capitalization consistent #26830

### LINKED ISSUE #26830: DOC: Make section title capitalization consistent
At the moment we've got a mix of capitalization formats in the title sections of our docs. For example, in the first level of the TOC tree in http://dev.pandas.io/ we've got (not the second word first letter, `s` lower case vs `G` upper case):
- `Getting started`
- `User Guide`

And this happens all across the documentation. I think it makes more sense to just capitalize the first letter of the first word, like in `Getting started`. More a personal preference than any objective reason, but looks simpler, more standard, and we avoid things like `pandas Ecosystem` which doesn't look really great to me.

CC: @TomAugspurger @jorisvandenbossche

### PR DESCRIPTION
**Pull request by restlessadmin**
- [X] closes #26830

This superseeds #26933, and should contain all the fixes requested there.

### CODE REVIEW

File: doc/source/getting_started/basics.rst [hunk 1]
```diff
@@ -643,7 +643,7 @@ there for details about accepted inputs.
 
 .. _basics.idxmin:
 
-Index of Min/Max Values
+Index of min/max Values
```
Comment (datapythonista) [inline doc/source/getting_started/basics.rst]: values in lowercase

**Review by datapythonista — COMMENTED** [review]
Looks good, there is just one of the cases from the previous review that wasn't changed.

File: doc/source/getting_started/basics.rst (hunk 1 above)
Comment (pmaxey83) [reply doc/source/getting_started/basics.rst]: Corrected that line.

**Review by datapythonista — APPROVED** [review]
Thanks @pmaxey83 great work here, will be nice to see the documentation rendered once this is merged (it'll be automatically published at dev.pandas.io)

Btw, for future contributions it'll be useful for you to use a separate branch for every PR you want to open. If you wanted to start working in something else now (until this is merged), you don't have an easy way, since you've got this in `master` (where other branches are created from).

**Comment by datapythonista** [comment]
@jorisvandenbossche merge when you're happy (and when you have time to review, will take a bit ;) )

File: doc/source/reference/series.rst [hunk 2]
```diff
@@ -119,7 +119,7 @@ Binary operator functions
    Series.product
    Series.dot
 
-Function application, GroupBy & Window
+Function application, GroupBy & window
```
Comment (jreback) [inline doc/source/reference/series.rst]: Groupby?

File: doc/source/user_guide/options.rst [hunk 3]
```diff
@@ -120,10 +120,10 @@ are restored automatically when you exit the `with` block:
    print(pd.get_option("display.max_columns"))
 
 
-Setting Startup Options in python/ipython Environment
+Setting startup options in Python/Ipython environment
```
Comment (jreback) [inline doc/source/user_guide/options.rst]: use IPython here

**Review by jreback — CHANGES_REQUESTED** [review]
couple of very small changes. ping on green.

File: doc/source/user_guide/advanced.rst [hunk 4]
```diff
@@ -179,18 +179,18 @@ on a deeper level.
 
 .. _advanced.shown_levels:
 
-Defined Levels
+Defined levels
 ~~~~~~~~~~~~~~
 
-The :class:`MultiIndex` keeps all the defined levels of an index, even
+The repr of a ``MultiIndex`` shows all the defined levels of an index, even
```
Comment (jorisvandenbossche) [inline doc/source/user_guide/advanced.rst]: I don't think this is a fully correct change. It is indeed shown in the repr, but here I think it was emphasized that the actual object keeps those.

File: doc/source/user_guide/advanced.rst [hunk 5]
```diff
@@ -179,18 +179,18 @@ on a deeper level.
...
 if they are not actually used. When slicing an index, you may notice this.
 For example:
 
 .. ipython:: python
 
-   df.columns.levels  # original MultiIndex
+   df.columns  # original MultiIndex
```
Comment (jorisvandenbossche) [inline doc/source/user_guide/advanced.rst]: Can you revert this? 
I would keep it here strictly on capitalisation changes, that makes it easier to review.

I think the above is also not a correct change, as we just changed the repr of a MI, and it will no longer show all levels, so you really need to look at the `.levels` attribute

File: doc/source/user_guide/advanced.rst [hunk 6]
```diff
@@ -210,8 +210,7 @@ To reconstruct the ``MultiIndex`` with only the used levels, the
 
 .. ipython:: python
 
-   new_mi = df[['foo', 'qux']].columns.remove_unused_levels()
-   new_mi.levels
```
Comment (jorisvandenbossche) [inline doc/source/user_guide/advanced.rst]: also revert this one

File: doc/source/user_guide/cookbook.rst [hunk 7]
```diff
@@ -706,7 +706,7 @@ Apply
                         for ind, row in df.iterrows()})
    df_orgz
 
-`Rolling Apply with a DataFrame returning a Series
+`Rolling apply with a DataFrame returning a series
```
Comment (jorisvandenbossche) [inline doc/source/user_guide/cookbook.rst]:
```suggestion
`Rolling apply with a DataFrame returning a Series
```

File: doc/source/user_guide/options.rst [hunk 8]
```diff
@@ -120,10 +120,10 @@ are restored automatically when you exit the `with` block:
...
 -----------------------------------------------------
 
-Using startup scripts for the python/ipython environment to import pandas and set options makes working with pandas more efficient.  To do this, create a .py or .ipy script in the startup directory of the desired profile.  An example where the startup folder is in a default ipython profile can be found at:
+Using startup scripts for the Python/Ipython environment to import pandas and set options makes working with pandas more efficient.  To do this, create a .py or .ipy script in the startup directory of the desired profile.  An example where the startup folder is in a default ipython profile can be found at:
```
Comment (jorisvandenbossche) [inline doc/source/user_guide/options.rst]:
```suggestion
Using startup scripts for the Python/IPython environment to import pandas and set options makes working with pandas more efficient.  To do this, create a .py or .ipy script in the startup directory of the desired profile.  An example where the startup folder is in a default ipython profile can be found at:
```

File: doc/source/whatsnew/v0.10.0.rst [hunk 9]
```diff
@@ -490,7 +490,7 @@ Updated PyTables Support
 however, query terms using the prior (undocumented) methodology are unsupported. You must read in the entire
 file and write it out using the new format to take advantage of the updates.
 
-N Dimensional Panels (Experimental)
+N dimensional panels (experimental)
```
Comment (jorisvandenbossche) [inline doc/source/whatsnew/v0.10.0.rst]:
```suggestion
N dimensional Panels (experimental)
```

File: doc/source/whatsnew/v0.25.0.rst [hunk 10]
```diff
@@ -74,41 +74,9 @@ a dict to a Series groupby aggregation (:ref:`whatsnew_0200.api_breaking.depreca
 
 See :ref:`groupby.aggregate.named` for more.
 
-
-.. _whatsnew_0250.enhancements.multi_index_repr:
-
```
Comment (jorisvandenbossche) [inline doc/source/whatsnew/v0.25.0.rst]: Ah, this is probably something that went wrong with updating with latest master, as this was a recent PR that was merged. 
My previous comments about things you changed in the MultiIndex text is probably caused by that as well. 

I would do:

```
git fetch upstream
git merge upstream/master
```

in this branch.

**Comment by pmaxey83** [comment]
That makes sense, I was really confused by those changes to MultiIndex.  I
have ran the specified commands and will apply the suggested changes.

File: doc/source/user_guide/advanced.rst (hunk 4 above)
Comment (datapythonista) [inline doc/source/user_guide/advanced.rst]: Not sure if this is from a merge, but we avoid having unrelated changes in PRs, in case something goes wrong and we need to revert. If this comes from a merge that applies it, please revert to the original. If it's you proposing to change those, that's welcome, but please open a separate PR for anything that is not about fixing the capitalization of titles.

File: doc/source/user_guide/advanced.rst (hunk 5 above)
Comment (datapythonista) [inline doc/source/user_guide/advanced.rst]: same

File: doc/source/user_guide/advanced.rst [hunk 11]
```diff
@@ -179,18 +179,18 @@ on a deeper level.
...
 if they are not actually used. When slicing an index, you may notice this.
 For example:
 
 .. ipython:: python
 
-   df.columns.levels  # original MultiIndex
+   df.columns  # original MultiIndex
 
-   df[['foo','qux']].columns.levels  # sliced
+   df[['foo','qux']].columns  # sliced
```
Comment (datapythonista) [inline doc/source/user_guide/advanced.rst]: same

File: doc/source/user_guide/advanced.rst [hunk 12]
```diff
@@ -210,8 +210,7 @@ To reconstruct the ``MultiIndex`` with only the used levels, the
 
 .. ipython:: python
 
-   new_mi = df[['foo', 'qux']].columns.remove_unused_levels()
-   new_mi.levels
+   df[['foo', 'qux']].columns.remove_unused_levels()
```
Comment (datapythonista) [inline doc/source/user_guide/advanced.rst]: same

File: doc/source/user_guide/cookbook.rst [hunk 13]
```diff
@@ -1099,7 +1099,7 @@ HDFStore
 
 The :ref:`HDFStores <io.hdf5>` docs
 
-`Simple Queries with a Timestamp Index
+`Simple queries with a timestamp index
```
Comment (datapythonista) [inline doc/source/user_guide/cookbook.rst]: Timestamp and Index are class names, can you leave the capitalization of those please

File: doc/source/user_guide/groupby.rst [hunk 14]
```diff
@@ -321,7 +321,7 @@ Index level names may be supplied as keys.
 
 More on the ``sum`` function and aggregation later.
 
-Grouping DataFrame with Index Levels and Columns
+Grouping DataFrame with index levels and columns
```
Comment (datapythonista) [inline doc/source/user_guide/groupby.rst]: Can you leave `Index` with uppercase

File: doc/source/user_guide/options.rst (hunk 3 above)
Comment (datapythonista) [inline doc/source/user_guide/options.rst]: `IPython` with capital P too

File: doc/source/whatsnew/v0.21.0.rst [hunk 15]
```diff
@@ -507,7 +507,7 @@ The configuration option ``pd.options.mode.use_inf_as_null`` is deprecated, and
 
 .. _whatsnew_0210.api_breaking.iteration_scalars:
 
-Iteration of Series/Index will now return Python scalars
+Iteration of series/index will now return Python scalars
```
Comment (datapythonista) [inline doc/source/whatsnew/v0.21.0.rst]: `Series/Index` with capital S and I

File: doc/source/whatsnew/v0.25.0.rst (hunk 10 above)
Comment (datapythonista) [reply doc/source/whatsnew/v0.25.0.rst]: this is still missing

**Review by datapythonista — COMMENTED** [review]
Thanks @pmaxey83, there are still few unrelated changes possibly from a merge. Can you fix those, and the couple of cases in the review comments, so we can get this merged. Thanks!

**Comment by jorisvandenbossche** [comment]
@pmaxey83 it seems that again something went wrong with merging master, as there a couple of changes in here from recent other PRs. Not fully sure how that happened or so what to recommend doing. One option might be to start from a clean branch from master, and only cherry-pick the commits from this branch with actual doc changes.

**Review by datapythonista — APPROVED** [review]
@pmaxey83 I saw you were having trouble restoring this PR to a correct state without unrelated changes, so I fixed it for you. You can leave it with us, no need to push or change anything, should be ready to merge now.

@jreback @jorisvandenbossche can you have a last look and merge if everything looks good please?

**Review by jorisvandenbossche — APPROVED** [review]

**Comment by jorisvandenbossche** [comment]
@pmaxey83 Thanks a lot!
